ui/src/app/settings/components/repos-list/repos-list.tsx

```diff
@@ -237,272 +233,357 @@ export class ReposList extends React.Component<RouteComponentProps<any>, {connec
+                                </>
+                            )}
                         </div>
                     }>
-                    <DataLoader load={() => services.projects.list('items.metadata.name').then(projects => projects.map(proj => proj.metadata.name).sort())}>
-                        {projects => (
-                            <Form
-                                onSubmit={params => this.connectHTTPSRepo(params as NewHTTPSRepoParams)}
-                                getApi={api => (this.formApiHTTPS = api)}
-                                defaultValues={{type: 'git'}}
-                                validateError={(params: NewHTTPSRepoParams) => ({
-                                    url: (!params.url && 'Repository URL is required') || (this.credsTemplate && !this.isHTTPSUrl(params.url) && 'Not a valid HTTPS URL'),
-                                    name: params.type === 'helm' && !params.name && 'Name is required',
-                                    password: !params.password && params.username && 'Password is required if username is given.',
-                                    tlsClientCertKey: !params.tlsClientCertKey && params.tlsClientCertData && 'TLS client cert key is required if TLS client cert is given.'
-                                })}>
-                                {formApi => (
-                                    <form onSubmit={formApi.submitForm} role='form' className='repos-list width-control'>
-                                        <div className='white-box'>
-                                            <p>CONNECT REPO USING HTTPS</p>
-                                            <div className='argo-form-row'>
-                                                <FormField formApi={formApi} label='Type' field='type' component={FormSelect} componentProps={{options: ['git', 'helm']}} />
+                    {this.state.method === ConnectionMethod.SSH && (
+                        <DataLoader load={() => services.projects.list('items.metadata.name').then(projects => projects.map(proj => proj.metadata.name).sort())}>
+                            {projects => (
+                                <Form
+                                    onSubmit={params => this.connectSSHRepo(params as NewSSHRepoParams)}
+                                    getApi={api => (this.formApiSSH = api)}
+                                    defaultValues={{type: 'git'}}
+                                    validateError={(params: NewSSHRepoParams) => ({
+                                        url: !params.url && 'Repository URL is required'
+                                    })}>
+                                    {formApi => (
+                                        <form onSubmit={formApi.submitForm} role='form' className='repos-list width-control'>
+                                            <div className='white-box'>
+                                                <p>Choose your connection method:</p>
```

Comment:
Also related to the comment on line 311, the connection method section and the drop down is repeated three times, one for each connection type, at lines 327 here, and 398 and 493.  There should only be one 'instance' because it is common to all connection types.

Reply:
The current implementation looks like it will attempt to connect to the repo each time you select the connection type.  Try the HTTPS connection type and fill out the repo url, and then switch the connection type (twice).